### ISSUE TITLE
Eliminate GLOB from CMake files

### ISSUE THREAD

**Issue by tkuchida** [opensim-org/opensim-core#787]
[GLOB is not recommended by CMake](https://cmake.org/cmake/help/v3.4/command/file.html).

**Comment by sherm1** [comment]
The alternative (listing all files) is pretty horrible for a large project though. It also causes added files to get missed. The main advantage (as I understand it) is to prevent any surprise additions of junk source files, but we normally handle that by directory structure (that is, don't put junk in a real src directory).

**Comment by tkuchida** [comment]
I don't have an opinion on this; it came up in #786. @klshrinidhi, please weigh in and close the issue if deemed appropriate.

**Comment by tkuchida** [comment]
Some pros and cons are discussed on Stack Overflow [here](http://stackoverflow.com/questions/1027247/best-way-to-specify-sourcefiles-in-cmake). One of the Stackers presents an interesting perspective (though I don't know whether this is still true with the latest tools). If you're a developer adding/removing files, globbing is better because you just need to re-run CMake to pick up your changes. For everyone else, it's better if the files are listed explicitly because then the developer adding/removing files would have had to modify CMakeLists.txt, so the changes would be detected and would get picked up automatically by their build system (rather than the user first getting a potentially obscure error when building fails and needing to know to try running CMake again---or wasting time figuring it out).

TL;DR: It seems that explicit file lists means more work for someone adding/removing files, but may help prevent issues for other users.

**Comment by sherm1** [comment]
I'm not aware of any time we've encountered the alleged problem with user builds, despite using GLOB for 10 years or so for Simbody & OpenSim. When does this happen? I'm wondering if this is a solution looking for a problem, at least in our case.

**Comment by elen4** [comment]
Whoops just saw this! @sherm1 I experienced failing builds when pulling code and trying to compile it without running CMake first, as I usually do because I am lazy :sweat_smile: and expect CMake to be automatically triggered if needed. I immediately recognized the issue because this happened to me in the past with other projects, but this is why I mentioned it in the discussion for #395. Apparently all of you have a different workflow than mine, so I would say this has low priority in the end (which is why I "gave up" in #395). If people start complaining about failing builds, you'll know where to start looking :grin:
